src/mrpro/operators/LinearOperatorMatrix.py

```diff
@@ -0,0 +1,357 @@
+"""Linear Operator Matrix class."""
+
+from __future__ import annotations
+
+import operator
+from collections.abc import Callable, Iterator, Sequence
+from functools import reduce
+from types import EllipsisType
+from typing import cast
+
+import torch
+from typing_extensions import Self, Unpack
+
+from mrpro.operators.LinearOperator import LinearOperator, LinearOperatorSum
+from mrpro.operators.Operator import Operator
+from mrpro.operators.ZeroOp import ZeroOp
+
+_SingleIdxType = int | slice | EllipsisType | Sequence[int]
+_IdxType = _SingleIdxType | tuple[_SingleIdxType, _SingleIdxType]
+
+
+class LinearOperatorMatrix(Operator[Unpack[tuple[torch.Tensor, ...]], tuple[torch.Tensor, ...]]):
+    r"""Matrix of Linear Operators.
+
+    A matrix of Linear Operators, where each element is a Linear Operator.
+
+    This matrix can be applied to a sequence of tensors, where the number of tensors should match
+    the number of columns of the matrix. The output will be a sequence of tensors, where the number
+    of tensors will match the number of rows of the matrix.
+    The j-th output tensor is calculated as
+    :math:`\sum_i \text{operators}[i][j](x[i])` where :math:`\text{operators}[i][j]` is the linear operator
+    in the i-th row and j-th column and :math:`x[i]` is the i-th input tensor.
```

Comment:
should be "in the j-th row and i-th column and :math:`x[i]` is the i-th input tensor."